crates/ty_python_semantic/src/types.rs

```diff
@@ -6460,6 +6477,10 @@ pub struct TypeVarInstance<'db> {
     /// The type var's definition (None if synthesized)
     pub definition: Option<Definition<'db>>,
 
+    /// The definition of the generic class, function, or type alias that binds this typevar. This
```

Comment:
It took me a while to understand the difference between `definition` and `binding_context`. I think it could help to add an example and add some comment explaining the difference between the two.